integration_tests/rollback/rollback_test.go

```diff
@@ -0,0 +1,82 @@
+package rollback_test
+
+import (
+	"context"
+	"github.com/metrumresearchgroup/command"
+	"github.com/sebdah/goldie/v2"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+const (
+	BaselineInstalled = "baseline-installed"
+	RollbackDisabledInstalled = "rollback-disabled-installed"
+)
+
+func TestRollback(t *testing.T) {
+	installCmd := command.New()
+	testCmd := command.New(command.WithDir("Rscripts"))
+	ctx := context.TODO()
+	// should have some setup to make sure the test-library is cleared out
+
+	_, err := installCmd.Run(ctx, "pkgr", "install", "--config=pkgr-baseline.yml")
+	if err != nil {
+		t.Fatalf("could not install baseline packages with err: %s", err)
+	}
+
+	t.Run("the baseline package was installed", func(t *testing.T) {
+		testRes, err := testCmd.Run(ctx, "Rscript", "--quiet", "install_test.R")
+		if err != nil {
+			t.Fatalf("failed to run Rscript command with err: %s", err)
+		}
+		g := goldie.New(t)
+		g.Assert(t, "baseline-installed", testRes.Output)
+	})
+
+	t.Run("will rollback on failure to install tarball", func(t *testing.T) {
+		res, err := installCmd.Run(ctx, "pkgr", "install", "--config=pkgr-rollback-tarball.yml", "--logjson")
+		if err != nil {
+			t.Fatalf("could not install baseline packages with err: %s", err)
+		}
+		rollbackOutputCheckHelper(t, res, BaselineInstalled)
+	})
+
+	t.Run("will not rollback on failure to install tarball when rollback disabled", func(t *testing.T) {
+		t.Run("in configuration file", func(t *testing.T) {
+			res, err := installCmd.Run(ctx, "pkgr", "install", "--config=pkgr-no-rollback-tarball.yml", "--logjson")
+			if err != nil {
+				t.Fatalf("could not install baseline packages with err: %s", err)
+			}
+			rollbackOutputCheckHelper(t, res, RollbackDisabledInstalled)
+		})
+
+		t.Run("as CLI flag", func(t *testing.T) {
+		    err := os.RemoveAll("test-library/ps")
```

Comment:
This feels a bit unstable to me. In general, I think it's better if each test is sort of working in a "fresh" environment and doesn't need bits of cleanup like this. Otherwise, we risk breaking this test when we change the order in which things happen.

I would instead consider adding a pkgr.yml file with a different test-library for tests like this. YMMV.

Reply:
i see where you're coming from, also debated that myself, the issue that popped up was basically... we have a ton of these combinations, and reconstructing the entire environment each time is going to be tedious. I was hoping to have some middle ground of cleanup when doing the same test multiple times, at least when the "cleanup" is simple. Now that I think about it though, this wouldn't actually need to completely re-fetch packages since it would actually pull from the previous steps install cache if present, but also if run in different order would still work and just do the entire pull. I think we can change this.

Reply:
@Dreznel see the commit below for an update - now i am resetting the entire library